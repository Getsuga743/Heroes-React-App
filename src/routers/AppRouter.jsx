import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import LoginScreen from '../components/Login/LoginScreen';
import DashboardRoutes from './DashboardRoutes';

const AppRouter = () => (
  <Router>
    {/* A <Switch> looks through its children <Route>s and
            renders the first one that matches the current URL. */}
    <Switch>
      <Route exact path="/login" component={LoginScreen} />
      <Route path="/" component={DashboardRoutes} />
    </Switch>
  </Router>
);

export default AppRouter;
