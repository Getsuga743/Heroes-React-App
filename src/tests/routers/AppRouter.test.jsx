import React from 'react';
import AppRouter from '../../routers/AppRouter';
import { mount } from 'enzyme';
import AuthContext from '../../auth/AuthContext';

describe('Pruebas en <AppRouter />', () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };
  test('debe de mostrar el login si no esta autenticado', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>,
    );
    expect(wrapper.find('h1').exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de mostrar el componente marvel si estÃ¡ autenticado', () => {
    const contextValueTrue = {
      dispatch: jest.fn(),
      user: {
        name: 'Ivan',
        logged: true,
      },
    };
    const wrapper = mount(
      <AuthContext.Provider value={contextValueTrue}>
        <AppRouter />
      </AuthContext.Provider>,
    );
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('nav').exists()).toBe(true);
  });
});
  