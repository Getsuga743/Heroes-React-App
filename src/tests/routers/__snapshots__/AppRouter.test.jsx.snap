// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el componente marvel si est√° autenticado 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </ul>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Hello Ivan
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
          type="button"
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div />,
]
`;

exports[`Pruebas en <AppRouter /> debe de mostrar el login si no esta autenticado 1`] = `
<div
  className="container mt-5 border"
>
  <h1>
    Login
  </h1>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
</div>
`;
